src/urllib3/_async/connectionpool.py

```diff
@@ -383,7 +383,13 @@ def _raise_timeout(self, err, url, timeout_value):
         host = self.host
         port = self.port
         scheme = self.scheme
-        request.add_host(host, port, scheme)
+
+        # When given a URL wit a trailing dot for the hostname part:
+        # "https://example.com./", urllib3 will strip off the dot and use the name
+        # without a dot internally and send it dot-less in HTTP Host: headers and in
+        # the TLS SNI field.
+
+        request.add_host(host.rstrip("."), port, scheme)
```

Comment:
Can you place `host = host.rstrip(".")` on its own line just below the comment? This way, we know that the comment only applies to host.rstrip, and not to request.add_host